diff --git a/admin_toolbar_tools/admin_toolbar_tools.module b/admin_toolbar_tools/admin_toolbar_tools.module
index 48ef363..1be5c98 100644
--- a/admin_toolbar_tools/admin_toolbar_tools.module
+++ b/admin_toolbar_tools/admin_toolbar_tools.module
@@ -459,12 +459,6 @@ function admin_toolbar_tools_menu_links_discovered_alter(&$links) {
       'menu_name' => 'admin',
       'parent' => 'admin_development',
     ];
-    $links['admin_toolbar_tools.devel.execute_php'] = [
-      'title' => t('Execute PHP Code'),
-      'route_name' => 'devel.execute_php',
-      'menu_name' => 'admin',
-      'parent' => 'admin_development',
-    ];
     $links['admin_toolbar_tools.devel.session'] = [
       'title' => t('Session viewer'),
       'route_name' => 'devel.session',
@@ -485,7 +479,15 @@ function admin_toolbar_tools_menu_links_discovered_alter(&$links) {
       'parent' => 'devel.admin_settings',
     ];
   }
-
+  // If module Devel_php is enabled.
+  if ($moduleHandler->moduleExists('devel_php')) {
+   $links['admin_toolbar_tools.devel_php.execute_php'] = [
+      'title' => t('Execute PHP Code'),
+      'route_name' => 'devel_php.execute_php',
+      'menu_name' => 'admin',
+      'parent' => 'admin_development',
+   ];
+  }
   // If module Views Ui enabled.
   if ($moduleHandler->moduleExists('views_ui')) {
     $links['admin_toolbar_tools.views_ui.add'] = [
@@ -613,3 +615,4 @@ function admin_toolbar_tools_installed_themes() {
 
   return $themes_installed;
 }
+
